api/api/utils/logger.py

```diff
@@ -0,0 +1,17 @@
+from typing import List, Dict, Union
+
+import api
+from api.api.models import APISource, APILog
+
+
+class APILogger:
+    @staticmethod
+    def log(
```

Comment:
we kinda don't need this but I see the vision.

tldr this is basically a "repository service" pattern, but Django models already are basically repository patterns.

basically, if you want this you can just add an extra method to the `APILog` model itself

Reply:
ok, I've moved the function to the APILog model and checked that it still works